src/NodeApi/JSValue.cs

```diff
@@ -391,29 +391,58 @@ public unsafe BigInteger ToBigInteger()
 
     public bool IsBigInt() => TypeOf() == JSValueType.BigInt;
 
-    public bool Is<TValue>() where TValue : struct
+    /// <summary>
+    /// Checks if the T struct can be created from this `JSValue`.
+    /// </summary>
+    /// <typeparam name="T">A struct that implements IJSValue interface.</typeparam>
+    /// <returns>
+    /// `true` if the T struct can be created from this `JSValue`. Otherwise it returns `false`.
+    /// </returns>
+    public bool Is<T>() where T : struct, IJSValue<T>
 #if NET7_0_OR_GREATER
-        , IJSValue<TValue> => TValue.CanBeConvertedFrom(this);
+        => T.CanCreateFrom(this);
 #else
-        => IJSValueShim<TValue>.CanBeConvertedFrom(this);
+        => IJSValueShim<T>.CanCreateFrom(this);
 #endif
 
-    public TValue? As<TValue>() where TValue : struct
-#if NET7_0_OR_GREATER
-        , IJSValue<TValue>
-        => TValue.CanBeConvertedFrom(this) ? TValue.CreateUnchecked(this) : default;
-#else
-        => IJSValueShim<TValue>.CanBeConvertedFrom(this)
-            ? IJSValueShim<TValue>.CreateUnchecked(this) : default;
-#endif
+    /// <summary>
+    /// Tries to create a T struct from this `JSValue`.
+    /// It returns `null` if the T struct cannot be created.
+    /// </summary>
+    /// <typeparam name="T">A struct that implements IJSValue interface.</typeparam>
+    /// <returns>
+    /// Nullable value that contains T struct if it was successfully created
+    /// or `null` if it was failed.
+    /// </returns>
+    public T? As<T>() where T : struct, IJSValue<T>
+        => Is<T>() ? AsUnchecked<T>() : default(T?);
 
-    public TValue AsUnchecked<TValue>() where TValue : struct
+    /// <summary>
+    /// Creates a T struct from this `JSValue` without checking the enclosed handle type.
+    /// It must be used only when the handle type is known to be correct.
+    /// </summary>
+    /// <typeparam name="T">A struct that implements IJSValue interface.</typeparam>
+    /// <returns>T struct created based on this `JSValue`.</returns>
+    public T AsUnchecked<T>() where T : struct, IJSValue<T>
 #if NET7_0_OR_GREATER
-        , IJSValue<TValue> => TValue.CreateUnchecked(this);
+        => T.CreateUnchecked(this);
 #else
-        => IJSValueShim<TValue>.CreateUnchecked(this);
+        => IJSValueShim<T>.CreateUnchecked(this);
 #endif
 
+    /// <summary>
+    /// Creates a T struct from this `JSValue`.
+    /// It throws `InvalidCastException` in case of failure.
+    /// </summary>
+    /// <typeparam name="T">A struct that implements IJSValue interface.</typeparam>
+    /// <returns>T struct created based on this `JSValue`.</returns>
+    /// <exception cref="InvalidCastException">
+    /// Thrown when the T struct cannot be crated based on this `JSValue`.
+    /// </exception>
+    public T CastTo<T>() where T : struct, IJSValue<T>
+        => As<T>()
+        ?? throw new InvalidCastException("JSValue cannot be casted to target type.");
```

Comment:
Include the target type `typeof(T).Name` in the message.

Reply:
Done

Reply:
The message is not updated.

Reply:
Fixed in the latest commit.